includes/class-wc-cart.php

```diff
@@ -450,9 +449,9 @@ public function check_cart_item_validity() {
 		$return = true;
 
 		foreach ( $this->get_cart() as $cart_item_key => $values ) {
-			$_product = $values['data'];
+			$product = $values['data'];
 
-			if ( ! $_product || ! $_product->exists() || 'trash' === $_product->post->post_status ) {
+			if ( ! $product || ! $product->exists() || 'trash' === $product->post->post_status ) {
```

Comment:
I think calling `->post` will return a deprecation notice. `get_status` should work here.